paddle/fluid/platform/mkldnn_helper.h

```diff
@@ -148,8 +148,6 @@ inline void ClearMKLDNNCache(const platform::Place& place,
     platform::MKLDNNDeviceContext* dev_ctx =
         (platform::MKLDNNDeviceContext*)pool.Get(place);
     dev_ctx->ResetBlobMap(ptr);
-    platform::MKLDNNDeviceContext::tls().set_cur_paddle_data_layout(
-        paddle::framework::DataLayout::kNCHW);
```

Comment:
Hi, @jczaja In the `dev_ctx->ResetBlobMap(ptr);`, there is some judgment that if it is **not** going to clear cache, then do not clear, then layout may not necessarily be KNCHW. But here we still forcibly set all datalyout to KNCHW ? Why ?

Reply:
kNCHW is a default setting. So we reset settings to defaults. There is no problem with second iteration because (Naive)Executor->Run set NCHW/NHWC will analyze model and set NHWC/NCHW accordingly

Reply:
Hi, but `ResetBlobMap(void* ptr)` includes two cases, clearing cache, and NotClearingCache two cases. In ClearingCache case, kNCHW has been set. And in notClearningCache case, only thing should do it reduce the counter. But why after ResetBlobMap(void* ptr) is called, kNCHW is called again, I thought in notclearingcache case, only thing to do is reduce the counter.

Reply:
Sorry, I just notice that it has been removed. LGTM! Thanks !